mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1567,7 +1580,7 @@ else if ( operation instanceof RemoveTupleOperation ) {
 				operation = queue.poll();
 			}
 
-			flushInserts( inserts );
+			flushInserts( provider, inserts );
```

Comment:
Why do we pass the provider to the `flushInsert` method? Maybe we could just use the field in the method.

Reply:
Sorry, it was my mistake. I hadn't seen that the method `flushInserts` was static

Reply:
I don't think it has to be static, but in general I prefer to pass the parameters to the methods. It makes refactoring easier.  It also makes it simpler to figure out what a method needs. Sometimes you have methods with no parameters and then you discover that they use 30 different fields.
Obviously, this is not a strict rule, more a loose guide line.